### PR TITLE
Refacto PreferencesController to use annotations

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refactor Controller to use latest conventions about annotations and update access rules.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |  Symfony page "Configure > Shop Parameters > General" should behave like before. **Access rules** have been updated: you need READ rights to display the page and CREATE or UPDATE or DELETE rights to modify it.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12074)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/PreferencesController.php [hunk 1]
```diff
@@ -77,40 +78,24 @@ public function indexAction(Request $request, FormInterface $form = null)
...
-        );
+        ]);
     }
 
     /**
-     * Process the form.
-     *
      * @param Request $request
      *
+     * @AdminSecurity("is_granted(['read','update', 'create','delete'], request.get('_legacy_controller'))",
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/PreferencesController.php]: Space :trollface:

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/PreferencesController.php [hunk 2]
```diff
@@ -47,9 +48,9 @@ class PreferencesController extends FrameworkBundleAdminController
      * @param Request $request
      * @param FormInterface|null $form
      *
-     * @Template("@PrestaShop/Admin/Configure/ShopParameters/preferences.html.twig")
+     * @AdminSecurity("is_granted(['read', 'update', 'create', 'delete'], request.get('_legacy_controller'))",
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/PreferencesController.php]:
```suggestion
     * @AdminSecurity("is_granted(['read', 'update', 'create', 'delete'], request.get('_legacy_controller'))")
```

**Comment by matks** [comment]
Requested changes have been applied and commit ammended

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by mbadrani** [comment]
@matks error 500 on import page
image

**Comment by PierreRambaud** [comment]
Hey @mbadrani, you need to test "Configure > Shop Parameters > General" only here, and with / without access rules :)

**Comment by mbadrani** [comment]
@matks need rebase to fix the import page (error 500)

**Comment by mbadrani** [comment]
I know @PierreRambaud but after the pr itself I always test regressions around, Mat confirmed me that need a rebase to avoid false flags in my tests

**Comment by matks** [comment]
Dismissed QA review as PR has been updated

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @matks

**Pull request merged**